osu.Game.Rulesets.Osu/Replays/OsuAutoGenerator.cs

```diff
@@ -156,6 +156,9 @@ private void addHitObjectReplay(OsuHitObject h)
             // TODO: Shouldn't the spinner always spin in the same direction?
             if (h is Spinner)
             {
+                if ((h as Spinner).Duration == 0)
+                    return;
```

Comment:
can be at least re-done to not typecheck twice (perhaps even inlined into one conditional, but I haven't checked)

```suggestion
            if (h is Spinner spinner)
            {
                if (spinner.Duration < reactionTime)
                    return;
```